speechbrain/lobes/models/huggingface_transformers/discrete_ssl.py

```diff
@@ -284,3 +330,49 @@ def forward(
             org_tokens, SSL_layers, deduplicates, bpe_tokenizers
         )
         return org_tokens, org_embedding, processed_tokens
+
+    def decode(self, tokens, SSL_layers=None):
+        """Takes an input waveform and return its corresponding waveform.
+
+        Arguments
+        ---------
+        tokens : torch.Tensor
+            A (Batch, codes, layers) tensor of discrete units
+        SSL_layers: List[int]:
+            determine which layers of SSL should be used by the vocoder.
+
+        Returns
+        -------
+        waveforms: torch.tensor
+            Batch of mel-waveforms [batch, time]
+        """
+
+        all_layer_ids = self.ssl_layer_ids
+        num_clusters = self.num_clusters[0]
```

Comment:
This code assumes the number of clusters is the same across all layers but in the constructor you can give a list. This might lead to subtle bugs in the future if a model with varying number of clusters is introduced. It would be good to at least add an assert statement to check the assumption holds.

Reply:
I agree. I think we are no longer using a varying number of clusters across layers, so let's remove this part.